src/bp/ui-admin/src/Pages/Workspace/Users/CreateUserModal.tsx

```diff
@@ -1,205 +1,146 @@
+import { Button, FormGroup } from '@blueprintjs/core'
+import { AuthRole, AuthStrategyConfig, CreatedUser, WorkspaceUser, WorkspaceUserInfo } from 'common/typings'
+import React, { FC, useEffect, useState } from 'react'
 import { connect } from 'react-redux'
-import Select from 'react-select'
 import AsyncSelect from 'react-select/lib/AsyncCreatable'
-import { AppState } from 'src/reducers'
+import { BaseDialog, DialogBody, DialogFooter } from '~/Pages/Components/BaseDialog'
+import RoleDropdown from '~/Pages/Workspace/Users/RoleDropdown'
 
 import api from '../../../api'
-import { fetchRoles } from '../../../reducers/roles'
+import { fetchAvailableUsers } from '../../../reducers/user'
 
-type Props = {
+import AuthStrategyDropdown from './AuthStrategyDropdown'
+
+interface OwnProps {
   isOpen?: boolean
   toggleOpen?: () => void
   onUserAdded?: () => void
   onUserCreated?: (newUser: CreatedUser) => void
-} & ReturnType<typeof mapStateToProps> & { fetchRoles }
-
-type SelectOption<T> = { label: string; value: T; __isNew__?: boolean }
-
-interface State {
-  users: WorkspaceUser[]
-  roles: SelectOption<string>[]
-  strategies: SelectOption<string>[]
-
-  selectedUser?: SelectOption<WorkspaceUser>
-  selectedOption?: SelectOption<string>
-  selectedStrategy?: SelectOption<string>
-  selectedRole?: SelectOption<string>
-
-  displayStrategy: boolean
+  forcedRoleId?: string
 }
 
-class CreateUserModal extends Component<Props, State> {
-  private formEl: any
-
-  readonly state: State = {
-    roles: [],
-    users: [],
-    strategies: [],
-    displayStrategy: false
-  }
+interface DispatchProps {
+  fetchAvailableUsers: (roleId?: string) => void
+}
 
-  componentDidMount() {
-    // tslint:disable-next-line: no-floating-promises
-    this.loadStrategies()
-    // tslint:disable-next-line: no-floating-promises
-    this.loadUsers()
-    this.loadRoles()
-  }
+interface StateProps {
+  availableUsers: WorkspaceUserInfo[]
+}
 
-  componentDidUpdate(prevProps: Props) {
-    if (!prevProps.roles.length && this.props.roles.length) {
-      this.loadRoles()
-    }
-  }
+type Props = DispatchProps & StateProps & OwnProps
 
-  private loadRoles() {
-    if (!this.props.roles.length) {
-      return this.props.fetchRoles()
-    }
+interface UserOption {
+  label: string
+  value: WorkspaceUser
+  __isNew__?: boolean
+}
 
-    const roles = this.props.roles.map(x => ({ value: x.id, label: x.name }))
-    this.setState({ roles, selectedRole: roles[0] })
-  }
+export const CreateUserModal: FC<Props> = props => {
+  const [role, setRole] = useState<AuthRole>()
+  const [strategy, setStrategy] = useState<AuthStrategyConfig>()
+  const [selectedUser, setSelectedUser] = useState<UserOption>()
 
-  async loadStrategies() {
-    const { data } = await api.getAnonymous().get('/auth/config')
-    this.setState({ strategies: data.payload.strategies.map(x => ({ label: x.strategyId, value: x.strategyId })) })
-  }
+  useEffect(() => {
+    props.fetchAvailableUsers()
```

Comment:
why do you fetchUser when this dialog opens ?

Reply:
Because when you remove a user from the workspace, unless you fetch again users on open, the user will not be there in the auto complete box